documentation/bolt_project_directories.md

```diff
@@ -4,6 +4,16 @@ Bolt runs in the context of a project directory or a `Boltdir`. This directory c
 
 The project directory structure makes it easy to share Bolt code by committing the project directory to Git. You can then check different repositories of Bolt code into different directories in order to manage various applications.
 
+## Project directory names
+
+Project directory names must match the expression: `[a-z][a-z0-9_]*`. In other words, they
+can contain only lowercase letters, numbers, and underscores, and begin with a 
+lowercase letter.
+
+Avoid giving your project the same name as another module in your modulepath.
+You cannot give your project the same name as a core Bolt module such as
+`boltlib`.
```

Comment:
I'm not sure that this needs to be explicitly stated, especially since we don't actually list the core Bolt modules anywhere. If a user happens to use a conflicting name, Bolt will display an error telling them as much. If we do want this in the docs I think it'd be helpful to list all of the core modules.